### PR TITLE
Fixed misaligned temp view in Single Pass scan and RadixSort kernels

### PR DESCRIPTION
**Pull request by MoFtZ**
The misaligned allocation of the boundary value used in the Single Pass scan kernel is causing some of the Cuda Inclusive Scan unit tests to fail. In particular, Int64 and Double.

### CODE REVIEW

File: Src/ILGPU.Algorithms/TempViewManager.cs [hunk 1]
```diff
@@ -86,6 +87,24 @@ public ArrayView<T> Allocate<T>(int length)
...
                 throw new ArgumentOutOfRangeException(ParamName);
 
+            // Ensure correct alignment when allocating types larger than a single int.
+            // NB: Structs are assumed to be int-aligned.
+            if (typeof(T).IsPrimitive)
+            {
+                var sizeOfAllocateT = Interop.SizeOf<T>();
+                var sizeOfInt = Interop.SizeOf<int>();
+                var allocationByteOffset = NumInts * sizeOfInt;
+                if (sizeOfAllocateT > sizeOfInt &&
```
Comment (MoFtZ) [inline Src/ILGPU.Algorithms/TempViewManager.cs:97]: QUESTION: What is the correct alignment rule?

RadixSort kernels appear to work just fine, but had to be changed because it did not follow the alignment rule as defined in this code.

**Review by m4rs-mt — APPROVED** [review]
